ivy/functional/frontends/tensorflow/nn.py

```diff
@@ -123,3 +123,14 @@ def batch_normalization(x, mean, variance, offset, scale, variance_epsilon, name
     return x * ivy.astype(inv, x.dtype, copy=False) + ivy.astype(
         offset - mean * inv if offset is not None else -mean * inv, x.dtype
     )
+
+
+def max_pool1d(input, ksize, strides, padding, data_format='NWC', name=None):
+    return ivy.max_pool1d(
```

Comment:
Hi @Aditya-Komaravolu 
Thank you for your time to contribute to Ivy 🙂
This function is not implemented on Ivy functional API if you want to implement this function you can do it compositionally following the frontend guide on compositions here [Compositions](https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends.html#compositions)

Reply:
Hi @hirwa-nshuti, thank you for your feedback. So this function should be transferred to general.py?

Reply:
No, basically Ivy doesn't have `max_pool1d` function implemented. Implementing this function's behavior can be done using other ivy functions that's what the guide is explaining on how to implement frontend functions using compositions.

Reply:
Oh okay, I get it now. Can I tag you if I get stuck anywhere?

Reply:
Yeah no problem feel free to tag me🙂

Reply:
Hey @hirwa-nshuti, I am unable to find which existing function is a close substitute to max_pool. Can you help please?

Reply:
If you're unable to implement the function compositionally I can recommend you to leave this task and take the one that is suitable for you.
Thanks

Reply:
@hirwa-nshuti I still want to try. I have a doubt. 

Can I deploy a maxpool1d function at ivy/functional/backend/tensorflow/layes.py and then reference it to the ivy/functional/frontend/tensorflow/layers.py ? I guess that might sort out the issue. I tried a sample code on my laptop and it seems to work.

Reply:
The thing is as we are currently welcoming the Ivy functional API extension suggestion. You can suggest it the function here before going straight away with implementations [Extend Ivy Functional API](https://github.com/unifyai/ivy/issues/3856)

Reply:
Sorry my bad.

Reply:
And also if you want to implement it to be fully supported by Ivy you will need to implement it across all Ivy backends. Feel free to open a PR with the extension implementation.

Reply:
![image](https://user-images.githubusercontent.com/64011471/190420835-0de48b38-320f-45f5-97fb-2780a3a6492a.png)

This is what have tried.

Reply:
> 

So should I have to move this function to the extensions.py file and then request for a PR?

Reply:
> And also if you want to implement it to be fully supported by Ivy you will need to implement it across all Ivy backends. Feel free to open a PR with the extension implementation. You can refer to the extension guideline here [Ivy API Extensions](https://lets-unify.ai/ivy/contributing/4_open_tasks.html?highlight=extension#ivy-api-extensions)

So should I move this function to `extensions.py` and then request a new PR? Also, do I have to write tests for this new code?

Reply:
Yeah once you implement the function to be fully supported by all ivy backends you'll have to write tests for it.

Reply:
@hirwa-nshuti I will get back to this function once I make a successful PR. I want to work on cosine_similarity func. Shall I close this PR?

Reply:
Yeah you can close the PR no worries